plugins/secretstores/http/http.go

```diff
@@ -0,0 +1,228 @@
+		if err != nil {
+			return fmt.Errorf("setting up data transformation failed: %w", err)
+		}
+		h.transformer = e
+	}
+
+	// Setup the decryption infrastructure
+	h.decrypter, err = h.DecryptionConfig.CreateDecrypter()
+	if err != nil {
+		return fmt.Errorf("creating decryptor failed: %w", err)
+	}
+
+	return nil
+}
+
+// Get searches for the given key and return the secret
+func (h *HTTP) Get(key string) ([]byte, error) {
+	v, found := h.cache[key]
+	if !found {
+		return nil, errors.New("not found")
+	}
+
+	if h.decrypter != nil {
+		// We got binary data delivered in a string, so try to
+		// decode it assuming base64-encoding.
+		buf, err := base64.StdEncoding.DecodeString(v)
+		if err != nil {
+			return nil, fmt.Errorf("base64 decoding failed: %w", err)
+		}
+		return h.decrypter.Decrypt(buf)
+	}
+
+	return []byte(v), nil
+}
+
+// Set sets the given secret for the given key
+func (h *HTTP) Set(_, _ string) error {
+	return errors.New("setting secrets not supported")
+}
+
+// List lists all known secret keys
+func (h *HTTP) List() ([]string, error) {
+	keys := make([]string, 0, len(h.cache))
+	for k := range h.cache {
+		keys = append(keys, k)
+	}
+	return keys, nil
+}
+
+// GetResolver returns a function to resolve the given key.
+func (h *HTTP) GetResolver(key string) (telegraf.ResolveFunc, error) {
+	// Download and parse the credentials
+	if err := h.download(); err != nil {
+		return nil, err
+	}
+
+	resolver := func() ([]byte, bool, error) {
+		s, err := h.Get(key)
+		return s, false, err
+	}
+	return resolver, nil
+}
+
+func (h *HTTP) download() error {
+	// Get the raw data form the URL
+	data, err := h.query()
+	if err != nil {
+		return fmt.Errorf("reading body failed: %w", err)
+	}
+
+	// Transform the data to the expected form if given
+	if h.transformer != nil {
+		out, err := h.transformer.EvalBytes(data)
+		if err != nil {
+			return fmt.Errorf("transforming data failed: %w", err)
+		}
+		data = out
+	}
+
+	// Extract the data from the resulting data
+	if err := json.Unmarshal(data, &h.cache); err != nil {
+		var terr *json.UnmarshalTypeError
+		if errors.As(err, &terr) {
+			return fmt.Errorf("%w; maybe missing or wrong data transformation", err)
+		}
+		return err
+	}
+
+	return nil
+}
+
+func (h *HTTP) query() ([]byte, error) {
+	request, err := http.NewRequest(http.MethodGet, h.URL, nil)
+	if err != nil {
+		return nil, fmt.Errorf("creating request failed: %w", err)
+	}
+
+	for k, v := range h.Headers {
+		if strings.ToLower(k) == "host" {
+			request.Host = v
+		} else {
+			request.Header.Add(k, v)
+		}
+	}
+
+	if err := h.setRequestAuth(request); err != nil {
+		return nil, err
+	}
+
+	resp, err := h.client.Do(request)
+	if err != nil {
+		return nil, fmt.Errorf("executing request failed: %w", err)
+	}
+	defer resp.Body.Close()
+
+	responseHasSuccessCode := false
+	for _, statusCode := range h.SuccessStatusCodes {
+		if resp.StatusCode == statusCode {
+			responseHasSuccessCode = true
+			break
+		}
+	}
```

Comment:
if you only used 2xx we could avoid this and just check http.StatusOK and one less thing users would need to be concerned about in their config